### PR TITLE
Add weather app canvas

### PR DESCRIPTION
**Pull request by alexdefender**
### Weather app canvas

[Demo](https://alexdefender.github.io/weather-app-canvas/) | [Code base](https://github.com/alexdefender/weather-app-canvas)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/alexdefender/weather-app-canvas/src/Weather.js [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+const RAIN = "#356dee";
+const WIND = "#7e807d";
+const LIGHTING = "#ffcd00";
+const SNOW = "#ffffff";
+const ERROR_404 = "404";
+
+function render(ctx, { height, width, weather }) {
+  requestAnimationFrame(() => render(ctx, { width, height, weather }));
+  let start = performance.now();
+  const hours = new Date().getHours();
```
Comment (roman01la) [inline submissions/alexdefender/weather-app-canvas/src/Weather.js]: Creating an instance of an object 60 times a second could be a huge perf and memory waste. It'd be better to instantiate an object somewhere outside of the rendering function.

File: submissions/alexdefender/weather-app-canvas/src/Weather.js [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+  if (6 <= hours && hours <= 11) {
```
Comment (roman01la) [inline submissions/alexdefender/weather-app-canvas/src/Weather.js]: Shouldn't `hours <= 11` do the same here?

File: submissions/alexdefender/weather-app-canvas/src/Weather.js [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    ctx.fillStyle = MORNING_SKY;
+  } else if (12 <= hours && hours <= 20) {
```
Comment (roman01la) [inline submissions/alexdefender/weather-app-canvas/src/Weather.js]: Same as above

File: submissions/alexdefender/weather-app-canvas/src/Weather.js [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+        if (ctx.fillStyle === NIGHT_SKY) {
+          drawMoon(width / 2, 80, MOON);
+        } else {
+          drawSun(width / 2, 100, SUN);
+        }
+        drawClouds(width / 2, 140, CLOUD);
+        break;
+      case iconList.brokenClouds[0]:
+      case iconList.brokenClouds[1]:
+        console.log(true);
```
Comment (roman01la) [inline submissions/alexdefender/weather-app-canvas/src/Weather.js]: Redundant logging

File: submissions/alexdefender/weather-app-canvas/src/Weather.js (hunk 1 above)
Comment (alexdefender) [reply submissions/alexdefender/weather-app-canvas/src/Weather.js]: Ok thx, I didn't think about it.

**Comment by alexdefender** [comment]
@roman01la thx for reviews and please review yet.

**Comment by roman01la** [comment]
Great job!

**Pull request merged**